features/step_definitions/pickle_steps.rb

```diff
@@ -12,9 +13,26 @@
 
 # create models from a table
 Given(/^the following #{capture_plural_factory} exists?:?$/) do |plural_factory, table|
+        puts plural_factory, table
   create_models_from_table(plural_factory, table)
 end
 
+Given(/^a section as such:?$/) do |table|
+  step "the sections as such:", table
+end
+
+Given(/^the sections as such:?$/) do |table|
+  header = table.raw[0]
+  data = table.raw[1..-1]
+  data.each{ |row|
+    opts = header.zip(row).to_h
+    opts.each{ |key,value|
+      opts[key] = value.include?(']') ? value.split(/[\]\[,]/)[1..-1]:value
```

Comment:
instead of `value.include?`, consider using a regular expression to find the content between the start and end delimiters if they exist